netket/nn/symmetric_linear.py

```diff
@@ -134,15 +126,18 @@ def __call__(self, x: Array) -> Array:
         """
         dtype = jnp.promote_types(x.dtype, self.dtype)
         x = jnp.asarray(x, dtype)
-        # infer in_features and ensure input dimensions (batch, in_features*n_sites)
-        if x.ndim == 1:
-            x = jnp.expand_dims(x, 0)
-            in_features = 1
-        elif x.ndim == 2:
-            in_features = 1
-        else:
-            in_features = x.shape[-2]
-            x = x.reshape(x.shape[0], -1)
+        # infer in_features and ensure input dimensions (batch, in_features,n_sites)
+
+        # TODO: Deprecated: Eventually remove and error if less than 3 dimensions
+        if x.ndim < 3:
+            if x.ndim == 1:
+                x_new = jnp.expand_dims(x, (0, 1))
+            elif x.ndim == 2:
+                x_new = jnp.expand_dims(x, 1)
+            symm_input_warning(x, x_new, "DenseSymm")
+            x = x_new
```

Comment:
```suggestion
        # TODO: Deprecated: Eventually remove and error if less than 3 dimensions
        if x.ndim < 3:
            old_shape = x.shape
            if x.ndim == 1:
                x = jnp.expand_dims(x, (0, 1))
            elif x.ndim == 2:
                x = jnp.expand_dims(x, 1)
            symm_input_warning(old_shape, x.shape, "DenseSymm")
```
For some reason, garbage collection for this sort of thing optimises really badly, and it's memory intensive